### PR TITLE
document Mosaic

### PR DESCRIPTION
**Pull request by Fil**
related #1011
mosaic

cc: @jheer

### CODE REVIEW

File: docs/lib/nyc-taxi.parquet
Comment (mbostock) [inline docs/lib/nyc-taxi.parquet:1]: Can we use a data loader to make this, or at least include the script to make it alongside this cached output?

Comment (Fil) [reply docs/lib/nyc-taxi.parquet:1]: Absolutely. I just didn't want to document a data loader as part of this page, which has enough matter already.

It's not a hugely complicated data loader, except it needs to install the binary duckdb.
```sh
export TMPDIR="docs/.observablehq/cache"
export PATH=$TMPDIR:$PATH
duckdb :memory: << EOF
-- Load spatial extension
INSTALL spatial; LOAD spatial;

-- Project, following the example at https://github.com/duckdb/duckdb_spatial
CREATE TEMP TABLE rides AS SELECT
  pickup_datetime::TIMESTAMP AS datetime,
  ST_Transform(ST_Point(pickup_latitude, pickup_longitude), 'EPSG:4326', 'EPSG:32118') AS pick,
  ST_Transform(ST_Point(dropoff_latitude, dropoff_longitude), 'EPSG:4326', 'EPSG:32118') AS drop
FROM 'https://uwdata.github.io/mosaic-datasets/data/nyc-rides-2010.parquet';

-- Write output parquet file
COPY (SELECT
  HOUR(datetime) + MINUTE(datetime) / 60 AS time,
  ST_X(pick)::INTEGER AS px, -- extract pickup x-coord
  ST_Y(pick)::INTEGER AS py, -- extract pickup y-coord
  ST_X(drop)::INTEGER AS dx, -- extract dropff x-coord
  ST_Y(drop)::INTEGER AS dy  -- extract dropff y-coord
FROM rides
ORDER BY 2,3,4,5,1 -- optimize output size by sorting
) TO '$TMPDIR/trips.parquet' (COMPRESSION 'ZSTD', row_group_size 10000000);
EOF

cat $TMPDIR/trips.parquet >&1  # Write output to stdout
rm $TMPDIR/trips.parquet       # Clean up
```

Comment (mbostock) [reply docs/lib/nyc-taxi.parquet:1]: I think have the data-loaded checked in but not functional, even if undocumented, is a big help!

Comment (jheer) [reply docs/lib/nyc-taxi.parquet:1]: This looks great. Thanks for including it! I agree that the data loader is an important component of the example, even if non-functional in this particular deployment.

Comment (Fil) [reply docs/lib/nyc-taxi.parquet:1]: OK I'll check in a simple version that does not try to work in CI — with the expectation that duckdb is on the $PATH.

File: docs/lib/mosaic.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Mosaic
+
+[Mosaic](https://uwdata.github.io/mosaic/) is a system for linking data visualizations, tables, and input widgets, all leveraging a database (DuckDB) for scalable processing. With Mosaic, you can interactively visualize and explore millions —and even billions— of data points.
+
+The example below shows 1 million taxi rides pickup and dropoff points in NYC on Jan 1-3, 2010. The dataset is stored in a 8MB parquet file, originally generated with a data loader.
```
Comment (mbostock) [inline docs/lib/mosaic.md:5]:
```suggestion
The example below shows 1 million taxi rides pickup and dropoff points in New York City on Jan 1–3, 2010. The dataset is stored in a 8MB [Apache Parquet](./arrow#apache-parquet) file, generated with a data loader.
```

(Dropping the “originally” for simplicity. It’s still accurate to say that it was generated by a data loader even if it’s not currently being re-generated by the data loader every time you build.)

File: docs/lib/mosaic.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+${maps}
+
+${histogram}
+
+The map views for pickups and dropoffs are coordinated. You can also select an interval in the histogram to filter the maps.
+_What spatial patterns can you find?_
```
Comment (mbostock) [inline docs/lib/mosaic.md:12]:
```suggestion
The map views for pickups and dropoffs are coordinated. You can also select an interval in the histogram to filter the maps. _What spatial patterns can you find?_
```

File: docs/lib/nyc-taxi.parquet.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+duckdb :memory: << EOF
```
Comment (mbostock) [inline docs/lib/nyc-taxi.parquet.sh:1]: Should this have a shebang? One of these probably…

```suggestion
#!/usr/bin/env bash
```

```suggestion
#!/bin/sh
```

**Review by mbostock — APPROVED** [review]

File: docs/lib/nyc-taxi.parquet.sh (hunk 3 above)
Comment (Fil) [reply docs/lib/nyc-taxi.parquet.sh:1]: I don't think so; shebang are only used for .exe?

Comment (mbostock) [reply docs/lib/nyc-taxi.parquet.sh:1]: Ah, so even if you have a shebang, it’s ignored because we explicitly run it with `sh`?

Comment (Fil) [reply docs/lib/nyc-taxi.parquet.sh:1]: yes sh ignores the shebang and interprets the rest of the code directly

```sh
❯ chmod +x ./test.py
❯ cat ./test.py
#! /usr/bin/env python3
print("hello, world")
❯ bash ./test.py
./test.py: line 2: syntax error near unexpected token `"hello, world"'
./test.py: line 2: `print("hello, world")'
❯ ./test.py
hello, world
```

Comment (mbostock) [reply docs/lib/nyc-taxi.parquet.sh:1]: Thanks for educating me. 😄

**Comment by mbostock** [comment]
Waiting for #1015 to land and then we’ll update this.

**Pull request merged**